Dart/src/com/jetbrains/lang/dart/ide/formatter/DartSpacingProcessor.java

```diff
@@ -384,6 +388,25 @@ else if (type1 == REFERENCE_EXPRESSION) {
       return Spacing.createSpacing(0, 1, preserveNewline, true, 0);
     }
 
+    if (elementType == NAMED_ARGUMENT) {
+      if (type1 == COLON) {
+        return addSingleSpaceIf(true);
+      }
+      if (type2 == COLON) {
+        return noSpace();
+      }
+    }
+    if (elementType == TYPE_ARGUMENTS && type1 == LT || type2 == GT) {
```

Comment:
class A < B > is covered partly by this 'if' (and probably that's not what you expected to be fixed by this code).
At first I was afraid that type2 == GT check is dangerous for relational operator. It's kind of luck that it is not affected. I think it would be good to make code easier to understand by adding one more check in addition to type2 == GT, for example type2 == GT && elementType == SOMETHING.